### ISSUE TITLE
[Package Request] Gitlab

### ISSUE THREAD

**Issue by jpscharf** [SynoCommunity/spksrc#218]
What about packages for Gitorious (http://gitorious.org/) and Gitlab (http://gitlabhq.com/)?
##

**Comment by garyee** [comment]
+1

**Comment by lioujheyu** [comment]
Recently I setup my gitlab inside the debian-chroot in XPEnology on VM. It works like a charm. If you re interesting in it, I write a installation note [here](https://github.com/lioujheyu/syno-chroot-gitlab/blob/master/README.md). 

So, I get a crazy idea here. Is there any way we could implement such gitlab spk but actually install it inside debian-chroot to avoid dependency hell?

**Comment by nszabadkai** [comment]
+1

**Comment by Decipher** [comment]
From https://www.synology.com/en-global/support/beta_dsm_5_2

> GitLab is a web-based Git repository manager with wiki and issue tracking features. It has numerous features and can be run on Synology NAS via Docker.

**Comment by Decipher** [comment]
Unfortunately for me, my NAS doesn't meet the hardware requirements to run Docker and therefore still can't run GitLab :-1:

**Comment by herve-loiret** [comment]
I use gitblit on my synology tomcat as a replacement and it work very well :-)

**Comment by jboxberger** [comment]
I've update the original package but less than 1GB of RAM makes GitLab to slow for normal usage.
https://github.com/jboxberger/synology-gitlab

**Comment by Finkman** [comment]
@jboxberger: I've installed Gitlab through the synologys package manager and ran into an issue:
The redisserver, that comes with the gitlab-package, is not configured and runs in verbose/debug mode. It writes unecessary log-infos to the disk every 5 minutes.
Can you help me? I couldn't get any response on the synology forums.

**Comment by jboxberger** [comment]
Maybe you can try my packgae (https://github.com/jboxberger/synology-gitlab/tree/master/bin) i use Gitlab 8.3.2, the original Packe uses Gitlab 7.9.2.

Otherwise you can try to update the docker redis Package

First you need to remove the old redis package "docker rmi <package name>"
docker rmi sameersbn/redis

Next you need to install the redis package again 
docker pull quay.io/sameersbn/redis:latest

Hope this helps.

**Comment by Finkman** [comment]
Is it possible to directly upgrate my existing GitLab installation to your package? Is this, that the update.sh is meant for?


Juri Boxberger schrieb am 11.01.2016 13:32:

**Comment by jboxberger** [comment]
I can't give you the warranty but i think this could work. My Update Routine is different from the Original Package, I also modified the configuration Files, so its is a bit tricky.
There is also new a constant from Gitlab 7 to Gitlab 8. Before you can Update you need to accept the Packages from Any Source "Package Manager->Setting->Trust"

First of all
- Make a Bakup of Maria DB (don't forget the user and password) and 
- Make a Backup of Git-Lab Folder (/volume1/docker/gitlab/)

If something goes wrong you can re-install the original synology package, restore the database and the gitlab files and you're good.

You can try to Update this way. 
1) Save Mode
- Clone all your rpositories to your local maschine
- Uninstall original Gitlab Package
- Install new Gitlab Package
- Configure Gitlab
- Push all Repositories back 

2) Experimental Mode
- note the gitlab Database user and password from docker overwiew
- Update the package (importatn: please insert the correct user and password in the form during the update process)
- if it doesn't work try to Update again
- if it doesn't work uninstall the whole Package (keep the Gitlab-Data)
- install the new package again. 

Hope this helps.

**Comment by Dr-Bean** [comment]
This is a bug tracker for SynoCommunity packages and SynoCommunity package requests. This is not the place to discuss other types of packages: please continue that discussion elsewhere.

Thanks.

**Comment by Eothred** [comment]
First of all, the title of this request could potentially be changed since Gitorious does not exist anymore. Secondly, I noticed after I upgraded from my old DS211 to DS216+ that Gitlab is made available by Synology through Docker on some models: https://www.synology.com/en-us/dsm/app_packages/Docker-GitLab.

Not sure if that makes this request redundant. If a package was created by SynoCommunity then perhaps GitLab would be available on more models, so perhaps still worth while. Beyond me to give any input there as I do not know all that much about Docker and performance in the first place.
